src/nccl_ofi_scheduler.c

```diff
@@ -42,6 +42,16 @@ static inline int set_schedule_by_threshold(nccl_net_ofi_threshold_scheduler_t *
 	/* Number of stripes is atleast 1 for zero-sized messages and at most equal to num of rails */
 	int num_stripes =
 		(int)NCCL_OFI_MAX(1, NCCL_OFI_MIN(NCCL_OFI_DIV_CEIL(size, scheduler->min_stripe_size), (unsigned)num_rails));
+
+	/* Ensure num_stripes is the nearest lower power of 2 */
```

Comment:
This isn't what we agreed to implement?  Don't assume that we won't be stupid and do 6 NICs per GPU.

Reply:
Right. I think you can have a simple loop as follows to find the closest number of rails to `num_stripes` that `num_rails` divides in:
```
for (int i = num_stripes; i >= 0; i--) {
	if ((num_rails % i) == 0) {
		num_stripes = i;
		break;
	}
}
```

Nit: I suggest to export above together with the calculation of existing `num_stripes` to a small utility function `get_num_stripes()`.

Reply:
Thanks for the detailed feedback. I have moved the implementation to get_num_stripes to a separate function as advised.